File: backend_go/internal/service/user_service.go
package service

import (
	"context"
	"fmt"
	"net/http"

	"github.com/HeHHeyboi/Cafe_Management/backend/internal/auth"
	"github.com/HeHHeyboi/Cafe_Management/backend/internal/dto"
	"github.com/HeHHeyboi/Cafe_Management/backend/internal/model"
	"github.com/HeHHeyboi/Cafe_Management/backend/internal/repository"
	"github.com/google/uuid"
)

type UserService struct {
	Repo   repository.UserRepo
	Secret string
}

func (us UserService) CreateUser(ctx context.Context, userReq dto.UserRequest) (dto.UserResponse, error) {
	var err error
	var user model.User
	user.FromUserReq(userReq)
	user.UserID = uuid.New()
	user.Password, err = auth.HashPassword(&user.Password)
	if err != nil {
		return dto.UserResponse{}, err
	}

	err = us.Repo.CreateUser(ctx, user)
	if err != nil {
		return dto.UserResponse{}, err
	}

	return user.ToUserResponse(), nil
}

func (us UserService) GetAllUser(ctx context.Context) ([]dto.UserResponse, error) {
	users, err := us.Repo.GetAllUser(ctx)
	if err != nil {
		return []dto.UserResponse{}, err
	}

	var response []dto.UserResponse
	for _, user := range users {
		response = append(response, user.ToUserResponse())
	}
	return response, nil
}

func (us UserService) GetUserByID(ctx context.Context, userid string) (dto.UserResponse, error) {
	data, err := us.Repo.GetUserByID(ctx, userid)
	if err != nil {
		return dto.UserResponse{}, err
	}

	response := dto.UserResponse{
		UserID:   data.UserID,
		Fname:    data.Fname,
		Lname:    data.Lname,
		Email:    data.Email,
		Password: data.Password,
		Role:     data.Role,
	}
	return response, nil
}

func (us UserService) Login(ctx context.Context, req dto.LoginRequest) (*http.Cookie, error) {
	data, err := us.Repo.GetUserByEmail(ctx, req.Email)
	if err != nil {
		return nil, dto.LoginError{Email: req.Email}
	}

	ok := auth.ComparePassword(&req.Password, &data.Password)
	if !ok {
		return nil, dto.LoginError{Email: req.Email}
	}

	cookie, err := auth.CreateCookie("id", data.UserID.String(), us.Secret)
	if err != nil {
		return nil, fmt.Errorf("Create Cookie Error, %s", err.Error())
	}
	return cookie, nil
}

func (us UserService) DeleteAllUser(ctx context.Context) error {
	err := us.Repo.DeleteAllUser(ctx)
	if err != nil {
		return err
	}
	return nil
}

File: backend_java/src/main/java/com/CafeManagement/service/UserService.java
package com.CafeManagement.service;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;
import java.util.concurrent.ExecutionException;

import javax.servlet.http.Cookie;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.ApplicationContext;
import org.springframework.http.ResponseCookie;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.CafeManagement.auth.Auth;
import com.CafeManagement.auth.CookieAuth;
import com.CafeManagement.dto.LoginRequest;
import com.CafeManagement.dto.LoginResult;
import com.CafeManagement.dto.UserRequest;
import com.CafeManagement.dto.UserResponse;
import com.CafeManagement.model.User;
import com.CafeManagement.repo.UserRepo;
import com.CafeManagement.repo.UserRepo.CreateUserParams;

@Service
@Transactional
public class UserService {
	UserRepo repo;
	ApplicationContext context;

	@Autowired
	public UserService(UserRepo repo, ApplicationContext context) {
		this.repo = repo;
		this.context = context;
	}

	public void CreateUser(UserRequest req) throws Exception {
		CreateUserParams params = new UserRepo.CreateUserParams(req);
		UUID userId = UUID.randomUUID();
		params.setUserId(userId.toString());
		repo.CreateUser(params);
	}

	public void DeleteAllUser() throws Exception {
		repo.DeleteAllUser();
	}

	public List<UserResponse> GetAllUser() throws Exception {
		List<User> users = repo.GetAllUser();
		List<UserResponse> response = new ArrayList<>();
		for (var user : users) {
			response.add(new UserResponse(user));
		}
		return response;
	}

	public UserResponse GetUserByEmail(String email) throws Exception {
		return new UserResponse(repo.GetUserByEmail(email));
	}

	public UserResponse GetUserById(String id) throws Exception {
		return new UserResponse(repo.GetUserById(id));
	}

	private final String authCookieName = "id";

	public LoginResult UserLogin(LoginRequest req) throws Exception {
		User user = repo.GetUserByEmail(req.getEmail());
		if (user == null) {
			return new LoginResult(false, "Can't Find user", null);
		}

		if (!Auth.ComparePassword(req.getPassword(), user.getPassword())) {
			return new LoginResult(false, "Password is Incorrect", null);
		}

		ResponseCookie cookie = CookieAuth.CreateCookie(authCookieName, user.getUserId().toString(),
				(String) context.getBean("secret"));
		return new LoginResult(true, "Login Success", cookie);
	}
}
